### DISCUSSION TITLE
How do I dynamically change the syntax of a command?

### DISCUSSION THREAD

**Discussion by akipiko** [Minestom/Minestom#2362] [Help] [answered]
Like [the attribute command](https://minecraft.fandom.com/wiki/Commands/attribute), is there a way to change the syntax depending on the argument?
For example, when "/attribute @s minecraft.something base" is entered, get and set are suggested as the next arguments, and when "/attribute @s minecraft.something modifier" is entered,  I want add, remove and value to be suggested as the next arguments.
(I'm not English speaker, so used translator a little.)

**Comment by TheMode** [comment, marked as answer]
These do not seem to be "dynamic syntaxes". Just two different syntax where one has the `base` literal, and another the `modifier` literal

**Reply by akipiko** [reply]
Thank you!
I finally understand that `ArgumentLiteral` should be used in such cases!

```
public AttributeCommand(){
        super("attribute");

        var targetArg = ArgumentType.Entity("target").singleEntity(true);
        var attributeArg = ArgumentType.String("attribute");
        var doubleArg = ArgumentType.Double("value");
        
        var baseArg = ArgumentType.Literal("base");
        var setArg = ArgumentType.Literal("set");
        
        addSyntax((sender, context) -> {
            // attribute <target> <attribute> base set <value>
        }, targetArg, attributeArg, baseArg, setArg, doubleArg);

        var getArg = ArgumentType.Literal("get");

        addSyntax((sender, context) -> {
            // attribute <target> <attribute> base get
        }, targetArg, attributeArg, baseArg, getArg);
    }
```
